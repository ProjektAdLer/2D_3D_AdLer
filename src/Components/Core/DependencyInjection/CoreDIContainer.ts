import { LoadWorldController } from "../Presentation/React/LoadWorldButton/LoadWorldController";
import ILoadWorldController from "../Presentation/React/LoadWorldButton/ILoadWorldController";
import { Container } from "inversify";
import CORE_TYPES from "./CoreTypes";
import ISceneView from "../Presentation/Babylon/SceneManagement/ISceneView";
import SceneView from "../Presentation/Babylon/SceneManagement/SceneView";
import SceneViewModel from "../Presentation/Babylon/SceneManagement/SceneViewModel";
import ScenePresenter from "../Presentation/Babylon/SceneManagement/ScenePresenter";
import MainScene from "../Presentation/Babylon/SceneManagement/MainScene";
import ICreateSceneClass from "../Presentation/Babylon/SceneManagement/ICreateSceneClass";
import IScenePresenter from "../Presentation/Babylon/SceneManagement/IScenePresenter";
import infrastructureDIContainer from "./InfrastructureDIContainer";
import useCaseDIContainer from "./UseCases/UseCaseDIConatiner";
import PortsDIContainer from "./Ports/PortsDIContainer";
import BuilderDIContainer from "./Builders/BuilderDIContainer";
import INavigation from "../Presentation/Babylon/Navigation/INavigation";
import Navigation from "../Presentation/Babylon/Navigation/Navigation";
import NavigationConfiguration from "../Presentation/Babylon/Navigation/NavigationConfiguration";

var CoreDIContainer = new Container();

// Scene
CoreDIContainer.bind<ISceneView>(CORE_TYPES.ISceneView)
  .to(SceneView)
  .inSingletonScope();
CoreDIContainer.bind(SceneViewModel).toSelf().inSingletonScope();
CoreDIContainer.bind<IScenePresenter>(CORE_TYPES.IScenePresenter)
  .to(ScenePresenter)
  .inSingletonScope();

// Navigation
CoreDIContainer.bind<INavigation>(CORE_TYPES.INavigation)
  .to(Navigation)
  .inSingletonScope();
CoreDIContainer.bind(NavigationConfiguration).toSelf();

// Controllers
CoreDIContainer.bind<ILoadWorldController>(CORE_TYPES.ILoadWorldController)
  .to(LoadWorldController)
  .inSingletonScope();

// bind other CreateSceneClass here for testing puposes -MK
CoreDIContainer.bind<ICreateSceneClass>(CORE_TYPES.ICreateSceneClass).to(
  MainScene
);

CoreDIContainer.load(infrastructureDIContainer);
CoreDIContainer.load(useCaseDIContainer);
CoreDIContainer.load(PortsDIContainer);
CoreDIContainer.load(BuilderDIContainer);

export default CoreDIContainer;
