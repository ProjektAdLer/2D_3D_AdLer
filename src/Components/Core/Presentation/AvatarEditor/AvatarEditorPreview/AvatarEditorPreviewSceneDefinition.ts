import { inject, injectable } from "inversify";
import AbstractSceneDefinition from "../../Babylon/SceneManagement/Scenes/AbstractSceneDefinition";
import type IPresentationDirector from "../../PresentationBuilder/IPresentationDirector";
import BUILDER_TYPES from "~DependencyInjection/Builders/BUILDER_TYPES";
import {
  Color3,
  Color4,
  DirectionalLight,
  HemisphericLight,
  Mesh,
  MeshBuilder,
  SceneLoader,
  ShadowGenerator,
  StandardMaterial,
  Vector3,
} from "@babylonjs/core";
import { ShadowOnlyMaterial } from "@babylonjs/materials/shadowOnly/shadowOnlyMaterial";
import AvatarEditorPreviewCameraBuilder from "./AvatarEditorPreviewCamera/AvatarEditorPreviewCameraBuilder";

const modelLink = require("../../../../../Assets/3dModels/sharedModels/3DModel_Avatar_male.glb");

@injectable()
export default class AvatarEditorPreviewSceneDefinition extends AbstractSceneDefinition {
  constructor(
    @inject(BUILDER_TYPES.IPresentationDirector)
    private director: IPresentationDirector,
    @inject(BUILDER_TYPES.IAvatarEditorPreviewCameraBuilder)
    private cameraBuilder: AvatarEditorPreviewCameraBuilder,
  ) {
    super();
  }

  protected async initializeScene(): Promise<void> {
    this.scene.clearColor = new Color4(0.91, 0.945, 0.977);

    // Create Preview Camera
    this.director.build(this.cameraBuilder);

    // Lights
    const hemiLight = new HemisphericLight(
      "ambientLight",
      new Vector3(0, 1, 0),
      this.scene,
    );
    hemiLight.intensity = 0.5;

    const keyLight = new DirectionalLight(
      "keyLight",
      new Vector3(1, -1, 1),
      this.scene,
    );
    keyLight.intensity = 1.8;

    const fillLight = new DirectionalLight(
      "fillLight",
      new Vector3(-1, 0, -1),
      this.scene,
    );
    fillLight.intensity = 0.7;

    keyLight.parent = this.scene.activeCamera;
    fillLight.parent = this.scene.activeCamera;

    // Avatar Placeholder
    let avatar: Mesh[];
    await SceneLoader.ImportMeshAsync("", modelLink, "", this.scene).then(
      (result) => {
        avatar = result.meshes as Mesh[];
        avatar[0].position.y = -1;
      },
    );

    // Ground Plane
    const groundPlane = MeshBuilder.CreatePlane(
      "plane",
      { size: 20 },
      this.scene,
    );
    groundPlane.position.y = -1;
    groundPlane.rotation.x = Math.PI / 2;
    const shadowMat = new ShadowOnlyMaterial("shadowOnly", this.scene);
    shadowMat.activeLight = keyLight;
    groundPlane.material = shadowMat;
    groundPlane.receiveShadows = true;

    const podium = MeshBuilder.CreateCylinder(
      "podium",
      { diameterTop: 1, diameterBottom: 1, height: 0.05, tessellation: 64 },
      this.scene,
    );
    podium.position.y = -1;
    podium.material = new StandardMaterial("podiumMat", this.scene);
    (podium.material as StandardMaterial).diffuseColor = new Color3(
      0.5,
      0.5,
      0.5,
    );
    podium.receiveShadows = true;

    // Shadow Generator
    const shadowGenerator = new ShadowGenerator(1024, keyLight);
    avatar!.forEach((mesh) => {
      shadowGenerator.addShadowCaster(mesh);
    });
    shadowGenerator.useBlurExponentialShadowMap = true;
    shadowGenerator.blurScale = 10;
    shadowGenerator.darkness = 0.6;
  }
}
