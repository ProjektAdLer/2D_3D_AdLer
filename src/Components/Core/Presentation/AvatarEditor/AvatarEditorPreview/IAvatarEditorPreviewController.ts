export default interface IAvatarEditorPreviewController {
  onTurnLeftUp(): void;
  onTurnLeftDown(): void;

  onTurnRightDown(): void;
  onTurnRightUp(): void;

  onZoomInDown(): void;
  onZoomInUp(): void;

  onZoomOutDown(): void;
  onZoomOutUp(): void;
}
