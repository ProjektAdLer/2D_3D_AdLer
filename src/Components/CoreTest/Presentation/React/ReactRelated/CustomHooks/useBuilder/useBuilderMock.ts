import * as useBuilder from "~ReactComponents/ReactRelated/CustomHooks/useBuilder";

/**
 * Mocks the useViewModelControllerProvider hook.
 * @param mockedTupels The return value, which the mocked hook will return.
 *
 */
export default function useBuilderMock(
  mockedTupels: [viewModel: any, controller: any] = [undefined, undefined]
) {
  jest.spyOn(useBuilder, "default").mockReturnValue(mockedTupels);
}
