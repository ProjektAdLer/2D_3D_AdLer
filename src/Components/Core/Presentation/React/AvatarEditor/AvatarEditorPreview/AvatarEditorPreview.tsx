import BabylonCanvas from "../../../Babylon/SceneManagement/BabylonCanvas";
import AvatarEditorPreviewSceneDefinition from "./AvatarEditorPreviewSceneDefinition";
import tailwindMerge from "../../../Utils/TailwindMerge";

type AvatarEditorPreviewProps = {
  className?: string;
};

export default function AvatarEditorPreview(props: AvatarEditorPreviewProps) {
  return (
    <BabylonCanvas
      className={tailwindMerge("rounded-xl", props.className ?? "")}
      sceneDefinitionType={AvatarEditorPreviewSceneDefinition}
    />
  );
}
