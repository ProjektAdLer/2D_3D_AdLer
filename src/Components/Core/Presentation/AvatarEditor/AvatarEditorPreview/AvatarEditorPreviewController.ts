import CoreDIContainer from "~DependencyInjection/CoreDIContainer";
import SCENE_TYPES, {
  ScenePresenterFactory,
} from "~DependencyInjection/Scenes/SCENE_TYPES";
import { ArcRotateCamera, Nullable, Observer, Scene } from "@babylonjs/core";
import IAvatarEditorPreviewController from "./IAvatarEditorPreviewController";
import AvatarEditorPreviewViewModel from "./AvatarEditorPreviewViewModel";
import bind from "bind-decorator";
import AvatarEditorPreviewSceneDefinition from "./AvatarEditorPreviewSceneDefinition";
import IScenePresenter from "../../Babylon/SceneManagement/IScenePresenter";

export default class AvatarEditorPreviewController
  implements IAvatarEditorPreviewController
{
  private scenePresenter: IScenePresenter;
  private camera: ArcRotateCamera | null = null;
  private sceneObserver: Nullable<Observer<Scene>>;

  constructor(private viewModel: AvatarEditorPreviewViewModel) {
    const factory = CoreDIContainer.get<ScenePresenterFactory>(
      SCENE_TYPES.ScenePresenterFactory,
    );
    this.scenePresenter = factory(AvatarEditorPreviewSceneDefinition);
  }

  @bind
  onTurnLeftDown() {
    this.sceneObserver =
      this.scenePresenter.Scene.onBeforeAnimationsObservable.add(() => {
        this.getCamera().alpha += this.viewModel.alphaStep;
      });
  }

  @bind
  onTurnLeftUp(): void {
    this.removeSceneObserver();
  }

  @bind
  onTurnRightDown() {
    this.sceneObserver =
      this.scenePresenter.Scene.onBeforeAnimationsObservable.add(() => {
        this.getCamera().alpha -= this.viewModel.alphaStep;
      });
  }

  @bind
  onTurnRightUp(): void {
    this.removeSceneObserver();
  }

  @bind
  onZoomInDown() {
    this.sceneObserver =
      this.scenePresenter.Scene.onBeforeAnimationsObservable.add(() => {
        this.getCamera().radius -= this.viewModel.zoomStep;
      });
  }

  @bind
  onZoomInUp(): void {
    this.removeSceneObserver();
  }

  @bind
  onZoomOutDown() {
    this.sceneObserver =
      this.scenePresenter.Scene.onBeforeAnimationsObservable.add(() => {
        this.getCamera().radius += this.viewModel.zoomStep;
      });
  }

  @bind
  onZoomOutUp(): void {
    this.removeSceneObserver();
  }

  private getCamera(): ArcRotateCamera {
    // get camera on demand to avoid race conditions
    if (!this.camera)
      this.camera = this.scenePresenter.Scene.activeCamera as ArcRotateCamera;
    return this.camera;
  }

  private removeSceneObserver() {
    if (this.sceneObserver)
      this.scenePresenter.Scene.onBeforeAnimationsObservable.remove(
        this.sceneObserver,
      );
  }
}
