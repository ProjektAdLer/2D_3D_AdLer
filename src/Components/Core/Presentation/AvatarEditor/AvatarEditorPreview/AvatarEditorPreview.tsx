import BabylonCanvas from "../../Babylon/SceneManagement/BabylonCanvas";
import AvatarEditorPreviewSceneDefinition from "./AvatarEditorPreviewSceneDefinition";
import tailwindMerge from "../../Utils/TailwindMerge";
import StyledButton from "~ReactComponents/ReactRelated/ReactBaseComponents/StyledButton";
import useBuilder from "~ReactComponents/ReactRelated/CustomHooks/useBuilder";
import AvatarEditorPreviewViewModel from "./AvatarEditorPreviewViewModel";
import IAvatarEditorPreviewController from "./IAvatarEditorPreviewController";
import BUILDER_TYPES from "~DependencyInjection/Builders/BUILDER_TYPES";

type AvatarEditorPreviewProps = {
  className?: string;
};

export default function AvatarEditorPreview(props: AvatarEditorPreviewProps) {
  const [, controller] = useBuilder<
    AvatarEditorPreviewViewModel,
    IAvatarEditorPreviewController
  >(BUILDER_TYPES.IAvatarEditorPreviewBuilder);

  return (
    <div className={tailwindMerge(props.className ?? "flex")}>
      <BabylonCanvas
        className={"rounded-xl w-full"}
        sceneDefinitionType={AvatarEditorPreviewSceneDefinition}
      />
      <div className="absolute bottom-0 flex space-x-2 p-2">
        <StyledButton
          onMouseDown={controller.onTurnLeftDown}
          onMouseUp={controller.onTurnLeftUp}
        >
          {"<"}
        </StyledButton>
        <StyledButton
          onMouseDown={controller.onZoomInDown}
          onMouseUp={controller.onZoomInUp}
        >
          {"+"}
        </StyledButton>
        <StyledButton
          onMouseDown={controller.onZoomOutDown}
          onMouseUp={controller.onZoomOutUp}
        >
          {"-"}
        </StyledButton>
        <StyledButton
          onMouseDown={controller.onTurnRightDown}
          onMouseUp={controller.onTurnRightUp}
        >
          {">"}
        </StyledButton>
      </div>
    </div>
  );
}
