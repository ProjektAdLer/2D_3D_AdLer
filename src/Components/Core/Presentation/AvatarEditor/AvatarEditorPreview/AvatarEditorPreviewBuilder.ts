import { injectable } from "inversify";
import PresentationBuilder from "../../PresentationBuilder/PresentationBuilder";
import AvatarEditorPreviewViewModel from "./AvatarEditorPreviewViewModel";
import IAvatarEditorPreviewController from "./IAvatarEditorPreviewController";
import AvatarEditorPreviewController from "./AvatarEditorPreviewController";

@injectable()
export default class AvatarEditorPreviewBuilder extends PresentationBuilder<
  AvatarEditorPreviewViewModel,
  IAvatarEditorPreviewController,
  undefined,
  undefined
> {
  constructor() {
    super(
      AvatarEditorPreviewViewModel,
      AvatarEditorPreviewController,
      undefined,
      undefined,
    );
  }
}
