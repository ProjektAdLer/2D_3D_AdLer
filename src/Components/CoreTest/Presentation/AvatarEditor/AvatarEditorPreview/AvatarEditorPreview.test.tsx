import React from "react";
import { fireEvent, render } from "@testing-library/react";
import AvatarEditorPreview from "../../../../Core/Presentation/AvatarEditor/AvatarEditorPreview/AvatarEditorPreview";
import CoreDIContainer from "../../../../Core/DependencyInjection/CoreDIContainer";
import useBuilderMock from "../../React/ReactRelated/CustomHooks/useBuilder/useBuilderMock";
import { mock } from "jest-mock-extended";
import IAvatarEditorPreviewController from "../../../../Core/Presentation/AvatarEditor/AvatarEditorPreview/IAvatarEditorPreviewController";
import { Provider } from "inversify-react";

jest.mock(
  "../../../../Core/Presentation/Babylon/SceneManagement/BabylonCanvas",
  () => "MockedBabylonCanvas",
);
jest.mock(
  "../../../../Core/Presentation/AvatarEditor/AvatarEditorPreview/AvatarEditorPreviewSceneDefinition",
);

const controllerMock = mock<IAvatarEditorPreviewController>();

describe("AvatarEditorPreview", () => {
  beforeAll(() => {
    CoreDIContainer.snapshot();
  });

  afterAll(() => {
    CoreDIContainer.restore();
  });

  // ANF-ID: [EZZ0055]
  test("should render", () => {
    useBuilderMock([undefined, controllerMock]);

    const renderResult = render(
      <Provider container={CoreDIContainer}>
        <AvatarEditorPreview />,
      </Provider>,
    );

    expect(renderResult.container).toMatchSnapshot();
  });

  test("MouseDown on left turn button calls controller", () => {
    useBuilderMock([undefined, controllerMock]);

    const renderResult = render(
      <Provider container={CoreDIContainer}>
        <AvatarEditorPreview />,
      </Provider>,
    );

    const leftTurnButton = renderResult.getByTestId("left-turn-button");

    fireEvent.pointerDown(leftTurnButton);

    expect(controllerMock.onTurnLeftDown).toHaveBeenCalledTimes(1);
  });

  test("MouseUp on turn left button calls controller", () => {
    useBuilderMock([undefined, controllerMock]);

    const renderResult = render(
      <Provider container={CoreDIContainer}>
        <AvatarEditorPreview />,
      </Provider>,
    );

    const leftTurnButton = renderResult.getByTestId("left-turn-button");

    fireEvent.pointerUp(leftTurnButton);

    expect(controllerMock.onTurnLeftUp).toHaveBeenCalledTimes(1);
  });

  test("MouseDown on zoom in button calls controller", () => {
    useBuilderMock([undefined, controllerMock]);

    const renderResult = render(
      <Provider container={CoreDIContainer}>
        <AvatarEditorPreview />,
      </Provider>,
    );

    const zoomInButton = renderResult.getByTestId("zoom-in-button");

    fireEvent.pointerDown(zoomInButton);

    expect(controllerMock.onZoomInDown).toHaveBeenCalledTimes(1);
  });

  test("MouseUp on zoom in button calls controller", () => {
    useBuilderMock([undefined, controllerMock]);

    const renderResult = render(
      <Provider container={CoreDIContainer}>
        <AvatarEditorPreview />,
      </Provider>,
    );

    const zoomInButton = renderResult.getByTestId("zoom-in-button");

    fireEvent.pointerUp(zoomInButton);

    expect(controllerMock.onZoomInUp).toHaveBeenCalledTimes(1);
  });

  test("MouseDown on zoom out button calls controller", () => {
    useBuilderMock([undefined, controllerMock]);

    const renderResult = render(
      <Provider container={CoreDIContainer}>
        <AvatarEditorPreview />,
      </Provider>,
    );

    const zoomOutButton = renderResult.getByTestId("zoom-out-button");

    fireEvent.pointerDown(zoomOutButton);

    expect(controllerMock.onZoomOutDown).toHaveBeenCalledTimes(1);
  });

  test("MouseUp on zoom out button calls controller", () => {
    useBuilderMock([undefined, controllerMock]);

    const renderResult = render(
      <Provider container={CoreDIContainer}>
        <AvatarEditorPreview />,
      </Provider>,
    );

    const zoomOutButton = renderResult.getByTestId("zoom-out-button");

    fireEvent.pointerUp(zoomOutButton);

    expect(controllerMock.onZoomOutUp).toHaveBeenCalledTimes(1);
  });

  test("MouseDown on right turn button calls controller", () => {
    useBuilderMock([undefined, controllerMock]);

    const renderResult = render(
      <Provider container={CoreDIContainer}>
        <AvatarEditorPreview />,
      </Provider>,
    );

    const rightTurnButton = renderResult.getByTestId("right-turn-button");

    fireEvent.pointerDown(rightTurnButton);

    expect(controllerMock.onTurnRightDown).toHaveBeenCalledTimes(1);
  });

  test("MouseUp on right turn button calls controller", () => {
    useBuilderMock([undefined, controllerMock]);

    const renderResult = render(
      <Provider container={CoreDIContainer}>
        <AvatarEditorPreview />,
      </Provider>,
    );

    const rightTurnButton = renderResult.getByTestId("right-turn-button");

    fireEvent.pointerUp(rightTurnButton);

    expect(controllerMock.onTurnRightUp).toHaveBeenCalledTimes(1);
  });
});
