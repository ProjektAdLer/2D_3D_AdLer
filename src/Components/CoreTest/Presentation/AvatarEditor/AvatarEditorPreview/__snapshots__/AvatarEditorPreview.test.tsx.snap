// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AvatarEditorPreview should render 1`] = `
<div>
  <div
    class=""
  >
    <mockedbabyloncanvas
      class="rounded-xl w-full h-full"
    />
    <div
      class="absolute bottom-4 portrait:bottom-2 flex justify-center w-full space-x-2 p-2"
    >
      <div
        class=""
      >
        <button
          class="justify-center p-1 lg:w-16 lg:h-16 md:w-14 md:h-14 sm:w-10 sm:h-10 w-10 h-10 aspect-square flex items-center text-sm rounded-lg hover:cursor-pointer landscape:hover:bg-adleryellow lg:text-xl transition ease-in-out duration-75 active:translate-x-[1px] active:translate-y-[1px] active:border-b-2 active:border-r-2 active:border-transparent active:bg-adleryellow text-adlerdarkblue font-regular border-b-2 border-r-2 border-adlerdarkblue overflow-hidden box-border cursor-pointer bg-buttonbgblue"
          data-testid="left-turn-button"
        >
          &lt;
        </button>
      </div>
      <div
        class=""
      >
        <button
          class="justify-center p-1 lg:w-16 lg:h-16 md:w-14 md:h-14 sm:w-10 sm:h-10 w-10 h-10 aspect-square flex items-center text-sm rounded-lg hover:cursor-pointer landscape:hover:bg-adleryellow lg:text-xl transition ease-in-out duration-75 active:translate-x-[1px] active:translate-y-[1px] active:border-b-2 active:border-r-2 active:border-transparent active:bg-adleryellow text-adlerdarkblue font-regular border-b-2 border-r-2 border-adlerdarkblue overflow-hidden box-border cursor-pointer bg-buttonbgblue"
          data-testid="zoom-in-button"
        >
          +
        </button>
      </div>
      <div
        class=""
      >
        <button
          class="justify-center p-1 lg:w-16 lg:h-16 md:w-14 md:h-14 sm:w-10 sm:h-10 w-10 h-10 aspect-square flex items-center text-sm rounded-lg hover:cursor-pointer landscape:hover:bg-adleryellow lg:text-xl transition ease-in-out duration-75 active:translate-x-[1px] active:translate-y-[1px] active:border-b-2 active:border-r-2 active:border-transparent active:bg-adleryellow text-adlerdarkblue font-regular border-b-2 border-r-2 border-adlerdarkblue overflow-hidden box-border cursor-pointer bg-buttonbgblue"
          data-testid="zoom-out-button"
        >
          -
        </button>
      </div>
      <div
        class=""
      >
        <button
          class="justify-center p-1 lg:w-16 lg:h-16 md:w-14 md:h-14 sm:w-10 sm:h-10 w-10 h-10 aspect-square flex items-center text-sm rounded-lg hover:cursor-pointer landscape:hover:bg-adleryellow lg:text-xl transition ease-in-out duration-75 active:translate-x-[1px] active:translate-y-[1px] active:border-b-2 active:border-r-2 active:border-transparent active:bg-adleryellow text-adlerdarkblue font-regular border-b-2 border-r-2 border-adlerdarkblue overflow-hidden box-border cursor-pointer bg-buttonbgblue"
          data-testid="right-turn-button"
        >
          &gt;
        </button>
      </div>
    </div>
  </div>
  ,
</div>
`;
